### PR TITLE
Charleystran/saml to icn

### PR DESCRIPTION
**Pull request by charleystran**
## Description of change
This change is to add an API endpoint so that an ICN can be retrieved for a user from a saml response

## Testing done
Specs have been added for each of the acceptance criteria

## Acceptance Criteria (Definition of Done)
- [x] Accepts the above request, performs an mvi lookup, and returns the icn from mvi
- [x] Returns appropriate errors when information to lookup icn is missing
- [x] Returns 404 error when we fail to lookup from mvi
- [x] Returns appropriate error when MVI is unavailable

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] https://github.com/department-of-veterans-affairs/vets-contrib/issues/1015
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: modules/openid_auth/spec/requests/mvi_user_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        'x-va-first-name' => 'Edward',
+        'x-va-middle-name' => 'John',
+        'x-va-last-name' => 'Paget'
+      }
+    end
+    it 'should respond properly when MVI is down' do
+      allow_any_instance_of(Faraday::Connection).to receive(:post).and_raise(Faraday::TimeoutError)
+      get '/internal/auth/v0/mvi-user', nil, auth_headers
+      expect(response).to have_http_status(:not_found)
+    end
```
Comment (dvogel) [inline modules/openid_auth/spec/requests/mvi_user_request_spec.rb]: @edpaget in the ticket you wrote separate AC bullet points for not found versus other "appropriate" error. Do you have a suggestion for the most appropriate error? Any of 502, 503, 504 could apply but is there a pattern we've chosen for other APIs that we should stick to?

File: modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+      def highest_loa
+        request.headers['x-va-level-of-assurance'].to_i == 3 ? 3 : 1
+      end
+
+      def check_required_headers
+        raise Common::Exceptions::ParameterMissing, 'X-VA-SSN or X-VA-EDIPI' if missing_ssn_or_edipi
+        raise Common::Exceptions::ParameterMissing, 'x-va-level-of-assurance' if missing_loa
+        raise Common::Exceptions::ParameterMissing, 'x-va-user-email' if missing_email
+      end
```
Comment (dvogel) [inline modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb]: Should this controller be emitting a `Vary` header to ensure proper caching? I don't know of anything between saml-proxy and vets-api but I'd hate to put some sort of caching proxy between them in the future and introduce a bug.

File: modules/openid_auth/spec/requests/mvi_user_request_spec.rb (hunk 1 above)
Comment (edpaget) [reply modules/openid_auth/spec/requests/mvi_user_request_spec.rb]: I think 503 makes the most sense to me. I'm unsure if there is another pattern.

**Comment by edpaget** [comment]
Looks like the build is failing because there's not quite enough coverage for the new controller. Need any help to get the tests over the line?

**Pull request merged**